ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/resources/impl/ResourceManager.java

```diff
@@ -203,11 +203,15 @@ public ResourceManager(@Assisted DevMachine devMachine,
                         Project[] tmpProjects = copyOf(projects, projects.length + 1);
                         tmpProjects[projects.length] = project;
                         projects = tmpProjects;
-
-                        eventBus.fireEvent(new ResourceChangedEvent(new ResourceDeltaImpl(project, ADDED | DERIVED)));
                     }
                 }
 
+                java.util.Arrays.sort(projects);
+
+                for (Project project : projects) {
```

Comment:
Added comment.

We need to guarantee that order of returned projects will be sorted according to the `compareTo` method implementation.